docs_src/design/adr/core/0009-V2-EdgeX-CLI.md

```diff
@@ -0,0 +1,66 @@
+# EdgeX-CLI V2 Design
+ 
+
+## Status
+
+**Draft**
+
+## Context
+This ADR presents a technical plan for creation of a 2.0 version of the edgex-cli which supports the new V2 REST APIs developed as part of the Ireland release of EdgeX.
+Existing Behavior
+The latest version of edgex-cli  (1.0.1) only supports the V1 REST APIs and thus cannot be used with the Ireland (or newer) release of EdgeX.
+
+As the edgex-cli was developed organically over time, the current implementation has a number of bugs mostly involving a lack of consistent behavior, especially with respect to formatting of output.
+
+Other issues with the existing client include:
+
+- lack of tab completion
+- default output of commands is too verbose
+- verbose output sometime prevents use of jq 
+- static configuration file required (i.e. no registry support)
+- project hierarchy not conforming to best practice guidelines (For instance /cmd should just contain the main application for the project, not an implementation for each command - that should be in /internal/cmd)
+
+## History
+The original Hanoi V1 client was created by a team at VMWare which is no longer participating in the project. 
+Canonical will lead the development of the Ireland/Jakarta V2 client.
+
+## Decision
+
+1. Use standardized command-line args/flags
+
+| Argument/Flag      | Description |
+| ----------- | ----------- |
+| `-d`, `--debug`      | show additional output for debugging purposes (e.g. REST URL, request JSON, …). This command-line arg will replace -v, --verbose and will no longer trigger output of the response JSON (see -j, --json).       |
+| `-j`, `--json`   | output the raw JSON response returned by the EdgeX REST API and *nothing* else. This output mode is used for script-based usage of the client.    |
+| `--version`   | output the version of the client and if available, the version of EdgeX installed on the system.    |
+
+
+2. Restructure the Go code hierarchy to follow the [most recent recommended guidelines](https://github.com/golang-standards/project-layout)
+
+3. Take full advantage of the more advanced features of the underlying command-line library, [Cobra](https://github.com/spf13/cobra), such as tab-completion of commands.
+
+5. Implement support for use of the client via the API Gateway, including being able to connect to a remote EdgeX instance.
+
+6. Implement all required V2 endpoints as per current [Issue](https://github.com/edgexfoundry/edgex-cli/issues) list.
+
+7. Change the client binary name from `edgex-cli` to `edgex`
+
+8. Add support for the four common endpoints - `/ping`, `/metrics`, `/version`, `/status`
+
+9. Change the syntax of CLI commands and arguments so that they match better the underlying REST API, specifying the microservice and the endpoint. Examples:
+
+    - `edgex data ping`
+    - `edgex metadata version`
+    - `edgex metadata metrics`
+
```

Comment:
Let's discuss. If we do make this change, I'd suggest mentioning that the <service> could be an abbreviation or full name. Ex.

```
$ edgex (d)ata ping
$ edgex (m)etadata ping
```

I suppose tab-completion gives you this in a shell, but this would simplify scripting.

That said, we should prepare for some pushback on this, and if so, at least offer an alternative flags-based solution for the common (e.g. `metrics`, `ping`, ...) endpoints.